### PR TITLE
TST: Update org for pytest-doctestplus in devinfra

### LINKED ISSUE #15810: TST: PluginValidationError: Plugin 'pytest_filter_subpackage' for hook 'pytest_ignore_collect' with pytest-dev
Example log: https://github.com/astropy/astropy/actions/runs/7375363390/job/20170602594

Not seeing this in pytest 8.0rc but it shows up on pytest-dev, so must be a change that got merged after RC was out.

```
pluggy._manager.PluginValidationError: Plugin 'pytest_filter_subpackage' for hook 'pytest_ignore_collect'
hookimpl definition: pytest_ignore_collect(path, config)
Argument(s) {'path'} are declared in the hookimpl but can not be found in the hookspec
```

Related issues:

* https://github.com/pytest-dev/pytest/issues/11779
* https://github.com/astropy/pytest-filter-subpackage/issues/14

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to test:

* https://github.com/astropy/pytest-filter-subpackage/pull/15
* https://github.com/scientific-python/pytest-doctestplus/pull/236
* https://github.com/matplotlib/pytest-mpl/pull/219

Fixes #15810 

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
Argh, of course my forks do not have version tags. I'll have to figure out how to get around the version checks next week.

**Comment by pllim** [comment]
Okay, devinfra job is green here, so that means we need to patch not one, but three different plugins and get them released so our devinfra job can pass again. 🤪 

@bsipocz and @astrofrog , are you able to help merge/release the affected plugins? 🙏

**Comment by bsipocz** [comment]
Has to wait till Monday, we got a lovely ❄️ storm 😀

**Comment by pllim** [comment]
No big rush, @bsipocz ! Since this is pytest-dev (not the RC), I think we have a while before it hits mainstream. 🤞

**Comment by pllim** [comment]
Actually maybe releases are unnecessary at this point, but those PRs would still need to be merged for packages testing against pytest-dev.

**Comment by bsipocz** [comment]
Merged what I could, but as you said, I'm not planning to do a release just yet but wait as more breaking changes or deprecations may come in the 8.1.x dev cycle.

**Comment by pllim** [comment]
Thanks, @bsipocz !

File: tox.ini [hunk 1]
```diff
@@ -93,12 +93,12 @@ deps =
     # Latest developer version of infrastructure packages.
     devinfra: git+https://github.com/pytest-dev/pytest.git
     devinfra: git+https://github.com/astropy/extension-helpers.git
-    devinfra: git+https://github.com/astropy/pytest-doctestplus.git
+    devinfra: git+https://github.com/scientific-python/pytest-doctestplus.git
```
Comment (pllim) [inline tox.ini:96]: Note: This should go in no matter what. This is an oversight from when we moved pytest-doctestplus to Scientific Python.

**Comment by pllim** [comment]
Yay, devinfra job is green again! This PR now only updates the org for pytest-doctestplus.

**Pull request merged**